titus-common/src/main/java/com/netflix/titus/common/util/rx/ReactorExt.java

```diff
@@ -162,6 +162,45 @@ public static void safeDispose(Disposable... disposables) {
         return TimerWithWorker.timer(action, worker, delay);
     }
 
+    /**
+     * In case subscriber does not provide exception handler, the error is propagated back to source, and the stream is broken.
+     * In some scenarios it is undesirable behavior. This method wraps the unprotected flux, and logs all unhandled
+     * exceptions using the provided logger.
+     */
+    public static <T> Flux<T> protectFromMissingExceptionHandlers(Flux<T> unprotectedStream, Logger logger) {
+        return Flux.create(sink -> {
+            Disposable disposable = unprotectedStream.subscribe(
+                    event -> {
+                        try {
+                            sink.next(event);
+                        } catch (Exception e) {
+                            try {
+                                sink.error(e);
+                            } catch (Exception e2) {
+                                logger.warn("Subscriber threw an exception from onNext handler", e);
```

Comment:
Shouldn't this line be in the try before the `sink.error` or before the try in general?

Reply:
Fixed.